### PR TITLE
FOGL-1180-1176 - Restapi end point for syslog + fix for large syslog size

### PR DESCRIPTION
**Pull request by amarendra-dianomic**
- [x] unit test added as per the requirements, and all are passing -- result (see next)

- [x] Works well on Raspbian with https + auth mandatory

- [x] Works well with https + auth optional

- [x] warning and error logs added

- [x] proper HTTP status code added for raise

**Test result:**

`=========================================================== 16 passed in 0.70 seconds ===========================================================
`
`==================================================== 794 passed, 12 skipped in 38.83 seconds ====================================================`

**Code Coverage:**

`Coverage for python/foglamp/services/core/api/support.py : 96% Show keyboard shortcuts
81 statements   78 run 3 missing 0 excluded`

### CODE REVIEW

**Comment by praveen-garg** [comment]
so no offset support ?

please remove syslogparser module, as its of no use now?

also attend feedback comments in https://github.com/foglamp/FogLAMP/pull/647 we could have used 647 and its branch itself cleanly without lossing feedback comments and new branch / PR

File: python/foglamp/services/core/api/support.py [hunk 1]
```diff
@@ -89,6 +90,45 @@
...
+        cmd = "grep -n '{}\[' {} | tail -n {}".format(valid_source[source.lower()], _SYSLOG_FILE, limit)
+        a = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE).stdout.readlines()
+        c = []
+        for b in a:
+            # Since "a" contains return value in bytes, convert it to string
+            c.append(b.decode())
+    except (OSError, Exception) as ex:
+        raise web.HTTPException(reason=str(ex))
+
+    return web.json_response({'logs': c, 'count': len(c)})
```
Comment (praveen-garg) [inline python/foglamp/services/core/api/support.py]: is c total count? per docstring and requirements

File: python/foglamp/services/core/support.py [hunk 2]
```diff
@@ -182,8 +179,11 @@ def add_machine_resources(self, pyz, file_spec):
     def add_psinfo(self, pyz, file_spec):
         # A PS listing of al the python applications running on the machine
         temp_file = self._interim_file_path + "/" + "psinfo-{}".format(file_spec)
-        a = os.popen('ps -eaf | grep python3').readlines()[:-2]
+        a = subprocess.Popen('ps -eaf | grep python3', shell=True, stdout=subprocess.PIPE).stdout.readlines()[:-2]
```
Comment (praveen-garg) [inline python/foglamp/services/core/support.py]: please add a comment why :-2 used

File: python/foglamp/services/core/support.py [hunk 3]
```diff
@@ -182,8 +179,11 @@ def add_machine_resources(self, pyz, file_spec):
...
+        c = []
+        for b in a:
+            c.append(b.decode())
```
Comment (praveen-garg) [inline python/foglamp/services/core/support.py]: `c = [b.decode() for b in a] `

File: python/foglamp/services/core/api/support.py [hunk 4]
```diff
@@ -89,6 +90,45 @@
...
+    except ValueError:
+        raise web.HTTPBadRequest(reason="{} is not a valid source".format(source))
+
+    try:
+        cmd = "grep -n '{}\[' {} | tail -n {}".format(valid_source[source.lower()], _SYSLOG_FILE, limit)
+        a = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE).stdout.readlines()
+        c = []
+        for b in a:
+            # Since "a" contains return value in bytes, convert it to string
+            c.append(b.decode())
```
Comment (praveen-garg) [inline python/foglamp/services/core/api/support.py]: `c = [b.decode() for b in a] `

**Comment by amarendra-dianomic** [comment]
@praveen-garg all feedback changes done. Pl check. Also, we never intended to add offset support. It was a copy/paste error. Now fixed.

**Comment by praveen-garg** [comment]
@MarkRiddoch  please see would it make sense to have rest end point to read syslogs? (its a proposal)

**Pull request merged**